client/app/dedicated/server/dedicated-server.service.js

```diff
@@ -687,18 +687,25 @@ angular
             });
         };
 
-        this.ipmiStartConnection = function (serviceName, type, ttl, ipToAllow, sshKey) {
-            return OvhHttp.post("/dedicated/server/{serviceName}/features/ipmi/access", {
+        this.ipmiStartConnection = function ({ serviceName, type, ttl, ipToAllow, sshKey, withGeolocation }) {
+            let promise = $q.when(ipToAllow);
+
+            if (withGeolocation) {
+                promise = this.getIpGeolocation().then((data) => _.get(data, "ip"));
+            }
+
+            return promise.then((ip) => OvhHttp.post("/dedicated/server/{serviceName}/features/ipmi/access", {
```

Comment:
You can also rename your param to: `ipToAllow`

Reply:
nope : 
error  'ipToAllow' is already declared in the upper scope  no-shadow